File: typescript/strongNumber.ts
export const strongNumber = (num: number): string => {
  const factorial = (n: number): number => {
    return n === 0 ? 1 : n * factorial(n - 1);
  };

  const sumOfFactorials = num
    .toString()
    .split('')
    .map(Number)
    .reduce((sum, digit) => sum + factorial(digit), 0);

  return sumOfFactorials === num ? "STRONG!!!!" : "Not Strong !!";
};

File: python/strong_number.py
def factorial(n):
    if n == 0:
        return 1
    return n * factorial(n - 1)

def strong_num(n):
    sum_of_factorials = sum(factorial(int(d)) for d in str(n))
    return "STRONG!!!!" if sum_of_factorials == n else "Not Strong !!"
